tubular/numeric.py

```diff
@@ -518,3 +525,190 @@ def transform(self, X):
+            will interpret ``svd_solver == 'auto'`` as ``svd_solver == 'full'``.
+            If ``0 < n_components < 1`` and ``svd_solver == 'full'``, select the
+            number of components such that the amount of variance that needs to be
+            explained is greater than the percentage specified by n_components.
+            If ``svd_solver == 'arpack'``, the number of components must be
+             strictly less than the minimum of n_features and n_samples.
+            Hence, the None case results in::
+                n_components == min(n_samples, n_features) - 1   svd_solver='auto', tol=0.0,  n_oversamples=10, random_state=None
+        svd_solver : {'auto', 'full', 'arpack', 'randomized'}, default='auto'
+            If auto :
+                The solver is selected by a default policy based on `X.shape` and
+                `n_components`: if the input data is larger than 500x500 and the
+                number of components to extract is lower than 80% of the smallest
+                dimension of the data, then the more efficient 'randomized'
+                method is enabled. Otherwise the exact full SVD is computed and
+                optionally truncated afterwards.
+            If full :
+                run exact full SVD calling the standard LAPACK solver via
+                `scipy.linalg.svd` and select the components by postprocessing
+            If arpack :
+                run SVD truncated to n_components calling ARPACK solver via
+                `scipy.sparse.linalg.svds`. It requires strictly
+                0 < n_components < min(X.shape)
+            If randomized :
+                run randomized SVD by the method of Halko et al.
+            .. sklearn versionadded:: 0.18.0
+
+        random_state : int, RandomState instance or None, default=None
+            Used when the 'arpack' or 'randomized' solvers are used. Pass an int
+            for reproducible results across multiple function calls.
+            .. sklearn versionadded:: 0.18.0
+    Attributes
+    ----------
+
+    pca : PCA class from sklearn.decomposition
+    n_components_ : int
+        The estimated number of components. When n_components is set
+        to 'mle' or a number between 0 and 1 (with svd_solver == 'full') this
+        number is estimated from input data. Otherwise it equals the parameter
+        n_components, or the lesser value of n_features and n_samples
+        if n_components is None.
+    feature_names_out: list or None
+        list of feature name representing the new dimensions.
+
+
+    """
+
+    def __init__(
+        self, columns, n_components=2, svd_solver="auto", random_state=None, **kwargs
+    ):
+
+        super().__init__(columns=columns, **kwargs)
+
+        if type(n_components) is int:
+            if n_components < 1:
+                raise ValueError(
+                    f"{self.classname()}:n_components must be positive  got {str(n_components)}"
```

Comment:
n_components must be >1

Reply:
Commit : [strictly positive condition](https://github.com/lvgig/tubular/pull/57/commits/90c146f635ec8f5c76a81977f4133a876864e356)